### PR TITLE
Honeycomb Refactor to Prep for otel

### PR DESCRIPTION
**Pull request by jfhbrook-at-work**
This PR is the current state of #101, minus the actual otel implementation. I ended up making a lot of important structural changes to the honeycomb abstractions, as well as tweaks to fix minor dev issues, and I wanted to consider them separately from the otel work. This should make it easier to review changes in the aggregate and get some much-needed fixes out there for downstream consumers.

This also includes a fix to the npm package so that it'll work on m1 macs.

Status: This passes boltzmann's unit tests, a local smoke test of echolibrium without honeycomb, and a CI test *with* honeycomb. I *also* was able to curl echolibrium and get traces:

image

I feel really good about this PR; I think it's Ready.

### CODE REVIEW

File: templates/boltzmann/core/honeycomb.ts [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    }
+
+    const schema = require('honeycomb-beeline/lib/schema')
+    const tracker = require('honeycomb-beeline/lib/async_tracker')
+    const _headerSources = headerSources || this.defaultHeaderSources
+    const traceContext = _getBeelineTraceContext(context)
+    const trace = beeline.startTrace({
+      [schema.EVENT_TYPE]: 'boltzmann',
+      [schema.PACKAGE_VERSION]: '1.0.0',
+      [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
```
Comment (ericsampson) [inline templates/boltzmann/core/honeycomb.ts]: per a discussion with ceej. It's worth verifying in your manual tests that this does the right thing, compared to what was being done before.
```suggestion
      [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}`,
```

Comment (jfhbrook-at-work) [reply templates/boltzmann/core/honeycomb.ts]: That change might've been an accident? 🤔 I can attempt at making this more testable though.

Comment (jfhbrook-at-work) [reply templates/boltzmann/core/honeycomb.ts]: Sorry, I misread this as a delta I already made, not a suggested change. I'm making it now, with a test.

Comment (jfhbrook-at-work) [reply templates/boltzmann/core/honeycomb.ts]: OK, change should've been made. Marking this resolved.

**Comment by jfhbrook-at-work** [comment]
I put some thought into this and I don't think it makes sense to review these changes separately from #101 when I'm effectively testing both sets of changes at the same time. Having talked about this with the team, I'm going to close this PR and focus on #101.

**Pull request closed without merging**